RealmSwift/SwiftUI.swift

```diff
@@ -1465,19 +1464,20 @@ private class ObservableAsyncOpenStorage: ObservableObject {
     private var appCancellable = [AnyCancellable]()
     private var asyncOpenCancellable = [AnyCancellable]()
 
-    @Published var asyncOpenState: AsyncOpenState = .connecting {
+    private var _state: AsyncOpenState = .connecting {
         willSet {
-            objectWillChange.send()
+            DispatchQueue.main.async {
```

Comment:
@tgoyne Because with `@AutoOpen` and `@AsyncOpen` we set the state directly from the view thread, (for example when there is no user logged when we change the state to waitingForUser), the warning is going to show even if we store it a different variable. We want to call `objectWillChange.send()` from the main thread to update the view, but we don't want to do it directly from the UI element.

Reply:
If the set is happening from within a call to update() I think we just don't want to trigger objectWillChange. If it's not, then we want to do the set immediately. Doing an async set causes problems for sendability and is just sort of spooky in general as it means we're temporarily in a weird inconsistent state.